### DISCUSSION TITLE
A query abou the task dependency

### DISCUSSION THREAD

**Discussion by JasonNing96** [v6d-io/v6d#787] [Q&A] [answered]
Hello, 
     I have some query about the tasks dependency of storage order; Like, If I have some much tasks and they are mutually dependent on each other, task2 should wait for the task1. How this work flow working, Whether V6D waits for the  task01 to complete and complete the storage before executing the task02, and whether V6D has related functions to detect whether the data is ready to complete?

**Comment by sighingnow** [comment, marked as answer]
Hi @JasonNing96,

(I have closed the issue to concentrate the discussion here.)

Vineyard itself is a storage engine, and doesn't aware about the workflow upon vineyard.

Vineyard does have some integration with workflow orchestration engines, e.g., airflow. In airflow, you would describe a task dependency between your task1 and task2, e.g., task2 requires task1. The workflow scheduler ensures task2 will be launched after task1 has been finished. Then when task2 been launched, the required input data should be ready as it has been produced by task1.

**Comment by JasonNing96** [comment]
Thanks!
